packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1017,7 +1038,11 @@ function performSyncWorkOnRoot(root) {
   const finishedWork: Fiber = (root.current.alternate: any);
   root.finishedWork = finishedWork;
   root.finishedLanes = lanes;
-  commitRoot(root);
+  if (enableSyncDefaultUpdates && includesSomeLane(lanes, DefaultLane)) {
```

Comment:
Needs to also check `DefaultHydrationLane`.

Though what I would do instead is `!includesSomeLane(lanes, SyncLane)`.